oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCheckCommand.java

```diff
@@ -242,16 +248,26 @@ static int checkDataStore(String... args) {
             }
 
             if (options.has(refOp) || options.has(consistencyOp)) {
-                if (options.has(verbose) &&
+
+                // Find blob ids by traversal for verbose mode + Segment Store or if verboseRootPath option
+                // is present (find blob references under a specific root path.)
+                if ((options.has(verbose) &&
                     (nodeStore instanceof SegmentNodeStore ||
-                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) {
+                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) || options.has(verboseRootPath)) {
                     NodeTraverser traverser = new NodeTraverser(nodeStore, dsType);
                     closer.register(traverser);
-                    traverser.traverse();
+
+                    if (options.has(verboseRootPath)) {
+                        List<String> rootPathList = options.valuesOf(verboseRootPath);
+                        traverser.traverse((String[]) rootPathList.toArray(new String[rootPathList.size()]));
```

Comment:
I think this case isn't needed.

Reply:
options.has(verboseRootPath) - this check ? Yes , makes sense, options.valuesOf(verboseRootPath) returns an empty list in case that option is not specified - I was thinking that this might throw an NPE or something